subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -101,28 +137,30 @@ public DefaultJvmTestSuite(String name, ConfigurationContainer configurations, D
             target.getTestTask().configure(task -> {
                 task.getTestFrameworkProperty().convention(getTestingFramework().map(framework -> {
                     switch(framework.framework) {
-                        case NONE:
-                        case JUNIT4:
+                        case NONE: // fall-through
+                        case JUNIT4: // fall-through
+                        case KOTLIN_TEST:
                             return new JUnitTestFramework(task, (DefaultTestFilter) task.getFilter());
-                        case JUNIT_JUPITER:
+                        case JUNIT_JUPITER: // fall-through
+                        case SPOCK:
```

Comment:
@big-guy Doesn't spock require the vintage engine in order to use JUnit Jupiter? Or no since it is Spock 2.0-groovy-3.0?

Reply:
Looks like the need for Vintage was removed with 2.0.